Wikipedia/Code/ArticleFullWidthImageCollectionViewCell.swift

```diff
@@ -1,17 +1,57 @@
-import UIKit
-
 @objc(WMFArticleFullWidthImageCollectionViewCell)
 open class ArticleFullWidthImageCollectionViewCell: ArticleCollectionViewCell {
-    override open var imageWidth: Int {
-        return traitCollection.wmf_leadImageWidth
-    }
     
-    override open class var nibName: String {
-        return "ArticleFullWidthImageCollectionViewCell"
+    override open func setup() {
+        let extractLabel = UILabel()
+        extractLabel.numberOfLines = 4
+        addSubview(extractLabel)
+        self.extractLabel = extractLabel
+        super.setup()
+        descriptionLabel.numberOfLines = 0
     }
+    
+    open override func sizeThatFits(_ size: CGSize, apply: Bool) -> CGSize {
+        let margins = UIEdgeInsetsMake(15, 13, 15, 13)
+        let spacing: CGFloat = 6
+        let saveButtonTopSpacing: CGFloat = 20
+        let widthMinusMargins = size.width - margins.left - margins.right
+        
+        var origin = CGPoint(x: margins.left, y: 0)
+        
+        if !isImageViewHidden {
+            if (apply) {
+                imageView.frame = CGRect(x: 0, y: 0, width: size.width, height: imageViewHeight)
+            }
+            origin.y += imageViewHeight
+        }
+        
+        origin.y += margins.top
+        
+        
+        let titleFrame = titleLabel.wmf_prefferedFrame(at: origin, fitting: widthMinusMargins, alignedBy: articleSemanticContentAttribute, apply: apply)
+        origin.y += titleFrame.height > 0 ? titleFrame.height + spacing : 0
```

Comment:
Could this height with spacing be DRY'ed to a rect extension? I see it's used a few places below...